### PR TITLE
SQLA disable warnings

### PR DESCRIPTION
**Pull request by guynir42**
Pass `overlaps` argument into `join_model` relationships and thus silence (but not fix) the SQLA warnings at app startup. 

I've fixed all the warnings but we should still investigate the underlying causes. 

Also, I've updated the cosmology loader to use the new methods, thus removing the astropy warning. 

This PR pins a branch of baselayer, when the PR on baselayer is merged, it should be changed to pin the main branch instead.

### CODE REVIEW

File: skyportal/models/obj.py [hunk 1]
```diff
@@ -249,15 +249,6 @@ class Obj(Base, conesearch_alchemy.Point):
         doc="Internal key used for secure websocket messaging.",
     )
 
-    candidates = relationship(
```
Comment (mcoughlin) [inline skyportal/models/obj.py:252]: Do we not use this anymore?

File: skyportal/models/spectrum.py [hunk 2]
```diff
@@ -385,8 +387,16 @@ def from_ascii(
         )
 
 
-SpectrumReducer = join_model("spectrum_reducers", Spectrum, User)
-SpectrumObserver = join_model("spectrum_observers", Spectrum, User)
+SpectrumReducer = join_model(
```
Comment (mcoughlin) [inline skyportal/models/spectrum.py:390]: This is clearly a big improvement.

**Review by mcoughlin — APPROVED** [review]
Just curious why the removal of candidates from the object table.

File: skyportal/models/obj.py (hunk 1 above)
Comment (guynir42) [reply skyportal/models/obj.py:252]: It is defined after the class code.

File: skyportal/models/spectrum.py (hunk 2 above)
Comment (guynir42) [reply skyportal/models/spectrum.py:390]: Yeah. I'm really surprised this thing even worked until now. The two relationships were overlapping and probably one killed the other.

Comment (mcoughlin) [reply skyportal/models/spectrum.py:390]: Yeah...

File: skyportal/models/obj.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/models/obj.py:252]: Ah ok.

**Comment by guynir42** [comment]
Test failures are all unrelated (the usual stuff). I think this could use a final review before merging @mcoughlin.

**Review by mcoughlin — APPROVED** [review]
LGTM!

**Pull request merged**